gateway/crates/gateway-binary/src/main.rs

```diff
@@ -59,7 +62,15 @@ fn main() -> anyhow::Result<()> {
         let crate_version = crate_version!();
         tracing::info!(target: GRAFBASE_TARGET, "Grafbase Gateway {crate_version}");
 
-        federated_server::serve(args.listen_address(), config, args.fetch_method()?, otel_tracing).await?;
+        federated_server::serve(ServerConfig {
```

Comment:
:zap: **Suggestion** 
Consider validating the `args` parameters before using them to ensure they meet expected formats or constraints, especially for `config_path` and `listen_addr`. This can prevent runtime errors.